src/libstore/globals.cc

```diff
@@ -189,6 +190,52 @@ template<> void BaseSetting<SandboxMode>::convertToArg(Args & args, const std::s
     });
 }
 
+template<> void BaseSetting<LogFormat>::set(const std::string & str)
+{
+    if (str == "raw")
+        value = LogFormat::raw;
+    else if (str == "raw-with-logs")
+        value = LogFormat::rawWithLogs;
+    else if (str == "internal-json")
+        value = LogFormat::internalJson;
+    else if (str == "bar")
+        value = LogFormat::bar;
+    else if (str == "bar-with-logs")
+        value = LogFormat::barWithLogs;
+    else throw UsageError("option '%s' has an invalid value '%s'", name, str);
+
+    createDefaultLogger();
+}
+
+template<> std::string BaseSetting<LogFormat>::to_string() const
+{
+    if (value == LogFormat::raw) return "raw";
+    else if (value == LogFormat::rawWithLogs) return "raw-with-logs";
+    else if (value == LogFormat::internalJson) return "internal-json";
+    else if (value == LogFormat::bar) return "bar";
+    else if (value == LogFormat::barWithLogs) return "bar-with-logs";
+    else abort();
+}
+
+template<> nlohmann::json BaseSetting<LogFormat>::toJSON()
+{
+    return AbstractSetting::toJSON();
+}
+
+template<> void BaseSetting<LogFormat>::convertToArg(Args & args, const std::string & category)
+{
+    args.addFlag({
```

Comment:
There's a way to define custom completers for the cli, you could use that so that the autocompletion suggest the different log formats

Reply:
I've done so (based on the completion function for `hashTypes`), but I don't know how to test if it really works (nor can I, AFAIK, as I use fish rather than bash or zsh).